Editor/EditorCore/MenuToggle.cs

```diff
@@ -0,0 +1,198 @@
+﻿
+#if UNITY_2019_1_OR_NEWER
+#define SHORTCUT_MANAGER
+#endif
+
+using System;
+using System.Diagnostics;
+using UnityEngine;
+using UnityEngine.ProBuilder;
+using Debug = UnityEngine.Debug;
+
+namespace UnityEditor.ProBuilder
+{
+    /// <summary>
+    /// Base class from which any action that is represented in the ProBuilder toolbar inherits.
+    /// </summary>
+    public abstract class MenuToggle: MenuAction
+    {
+        public enum MenuToggleState
+        {
+            Inactive,
+            Active,
+        };
+
+        MenuToggleState m_CurrentState;
+
+        protected delegate void StartEndCallBack();
+
+
+        protected MenuToggle()
+        {
+            iconMode = ProBuilderEditor.s_IsIconGui;
+            m_CurrentState = MenuToggleState.Inactive;
+
+            EditorApplication.update += OnUpdate;
+        }
+
+        public override void OnActionChanged(MenuAction action)
+        {
+            if(m_CurrentState == MenuToggleState.Active && !this.Equals(action))
+            {
+                Debug.Log("Changing menu action! Deactivating current toggle");
+                EndActivation(OnEnd);
+            }
+        }
```

Comment:
A general thought on `MenuToggleState`, could this be handled instead by defining an associated EditorTool and checking that it is active?